src/t_mockiato.rs

```diff
@@ -0,0 +1,367 @@
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo(|key| key.any());
+        mock.foo(2);
+    }
+
+    fn modules() {
+        // Only traits can be mocked
+        unimplemented!()
+    }
+
+    fn return_constant() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self) -> i16;
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo().returns(2i16);
+        assert_eq!(mock.foo(), 2);
+    }
+
+    // Mockiato implemented this feature only for one type: ()
+    fn return_default() {
+        unimplemented!()
+    }
+
+    fn return_panic() {
+        unimplemented!()
+    }
+
+    // This is the default behavior
+    fn times_once() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo().times(1);
+        mock.foo();
+    }
+
+    fn times_any() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo().times(..);
+        mock.foo();
+        mock.foo();
+    }
+
+    fn times_n() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo().times(2);
+        mock.foo();
+        mock.foo();
+    }
+
+    fn times_never() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo().times(0);
+    }
+
+    fn many_args() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self, a: i8, b: i8, c: i8, d: i8, e: i8, f: i8, g: i8,
+                   h: i8, i: i8, j: i8, k: i8, l: i8, m: i8, n: i8, o: i8,
+                   p: i8);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo(|x| x.any(), |x| x.any(), |x| x.any(), |x| x.any(),
+                        |x| x.any(), |x| x.any(), |x| x.any(), |x| x.any(),
+                        |x| x.any(), |x| x.any(), |x| x.any(), |x| x.any(),
+                        |x| x.any(), |x| x.any(), |x| x.any(), |x| x.any());
+        mock.foo(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15);
+        print!("≥ 16 ");
+    }
+
+    fn sequence() {
+        #[mockable]
+        pub trait A {
+            fn foo(&self, x: i32);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_foo(|x| x.partial_eq(42)).returns(());
+        mock.expect_foo(|x| x.partial_eq(5)).returns(());
+        mock.expect_foo_calls_in_order();
+        mock.foo(42);
+        mock.foo(5);
+        print!("single method ");
+    }
+
+    fn version() {
+        let ver = crate::built_info::DEPENDENCIES.iter()
+            .find(|(name, _)| *name == "mockiato")
+            .unwrap()
+            .1;
+        print!("{} ", ver);
+    }
+
+    fn where_clause() {
+        // I think where clauses work but only for generic traits, not generic
```

Comment:
Yes, where clauses work for generic traits:
https://github.com/myelin-ai/mockiato/blob/a1a06c8fea51f10bc22182b06a691b4ea2c91dfb/tests/generic_types.rs#L10